File: goblinframework-api/src/main/java/org/goblinframework/api/compression/Compressor.java
package org.goblinframework.api.compression;

public enum Compressor {

  BZIP2("bzip2"),
  GZIP("gz"),
  DEFLATE("deflate");

  private final String name;

  Compressor(String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }
}

File: goblinframework-api/src/main/java/org/goblinframework/api/compression/Compressor.kt
package org.goblinframework.api.compression

@Deprecated("To be removed")
enum class Compressor(val id: Byte, val algorithm: String) {

  BZIP2(1.toByte(), "bzip2"),
  GZIP(2.toByte(), "gz"),
  DEFLATE(3.toByte(), "deflate")

}
